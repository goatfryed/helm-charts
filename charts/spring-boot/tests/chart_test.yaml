suite: test chart
tests:
  - it: works with minimal configuration
    set:
      image.repository: goatfryed/spring-boot-echo
    asserts:
      - template: deployment.yaml
        matchSnapshot:
      - template: propertiesConfigMap.yaml
        matchSnapshot:
      - template: service.yaml
        matchSnapshot:
      - template: ingress.yaml
        matchSnapshot:
      - template: serviceaccount.yaml
        matchSnapshot:
      - template: hpa.yaml
        matchSnapshot: