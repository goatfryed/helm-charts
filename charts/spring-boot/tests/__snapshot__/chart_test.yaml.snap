works with minimal configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: spring-boot
        helm.sh/chart: spring-boot-0.1.0
      name: RELEASE-NAME-spring-boot
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: spring-boot
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: spring-boot
            helm.sh/chart: spring-boot-0.1.0
        spec:
          containers:
            - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: null
              image: goatfryed/spring-boot-echo:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: spring-boot
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /app/config/helm-config
                  name: helm-config
          securityContext: {}
          serviceAccountName: RELEASE-NAME-spring-boot
          volumes:
            - configMap:
                name: RELEASE-NAME-spring-boot-application-properties
              name: helm-config
  2: |
    apiVersion: v1
    data: null
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-spring-boot-application-properties
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: spring-boot
        helm.sh/chart: spring-boot-0.1.0
      name: RELEASE-NAME-spring-boot
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: spring-boot
      type: ClusterIP
  4: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: spring-boot
        helm.sh/chart: spring-boot-0.1.0
      name: RELEASE-NAME-spring-boot
